R/classification.R

```diff
@@ -0,0 +1,193 @@
+#' Retrieve the taxonomic hierarchies from local database
+#'
+#' This function is equivalent to the taxize::classification function, except
+#' that it uses a local database (so is much faster) and is currently
+#' restricted to handling NCBI taxa. The output is identical to
+#' taxize::classification.
+#'
+#' @param x Vector of taxon keys for the given database
+#' @param db The database to search
+#' @param verbose Print verbose messages
+#' @param ... Additional arguments passed to database specific classification functions.
+#' @return list of data.frames with the columns: name, rank, and id. This is
+#' exactly equivalent to the output of 'taxize::classification'. 
+#' @export
+#' @examples
+#' \dontrun{
+#' classification(c(3702, 9606))
+#' }
+classification <- function(x, db='ncbi', verbose=TRUE, ...){
+  lineages <- if(is.null(x) || length(x) == 0){
+    # For empty or NULL input, return empty list
+    list()
+  } else {
+    FUN <- switch(db,
+      itis = itis_classification,
+      tpl  = tpl_classification,
+      col  = col_classification,
+      gbif = gbif_classification,
+      ncbi = ncbi_classification,
+      stop("Database '", db, "' is not supported")
+    )
+    src <- autoload(db)
+    lineages <- FUN(src, x, ...)
+  }
+
+  attributes(lineages) <- list(names=names(lineages), class='classification', db=db)
+
+  if(verbose && all(is.na(lineages))){
+    message("Not found. Consider checking the spelling or alternative classification")
+  }
+
+  lineages
+}
+
+autoload <- function(db){
+  if(db == 'ncbi'){
+    src_ncbi(db_download_ncbi())
```

Comment:
@arendsee wondering if calling this on each `taxizedb::classification()` invocation will create a new DBI connection each time, and if there's any downside to that. perhaps there's not? maybe there is? not familiar enough with DBI, maybe on function exit the connection is gone since it's not being assigned to anything outside of the scope of `taxizedb::classification()` ?

Reply:
I think it opens a new connection. It is super fast, so performance isn't an issue.

I know SQLite allows multiple users. There can be conflicts when they are all trying to write to the database, but since we are only reading, it should be fine.

I should probably call `disconnect` when I *leave* the functions (I get warnings about that sometimes).

Reply:
> I should probably call disconnect when I leave the functions 

that sounds good